packages/dashboard/src/app/views/editorSettings/archive/index.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useCallback, useMemo } from '@web-stories-wp/react';
+import { v4 as uuidv4 } from 'uuid';
+import { __ } from '@web-stories-wp/i18n';
+import { useFeature } from 'flagged';
+import { Checkbox, THEME_CONSTANTS } from '@web-stories-wp/design-system';
+
+/**
+ * Internal dependencies
+ */
+import {
+  SettingForm,
+  SettingHeading,
+  CheckboxLabel,
+  CheckboxLabelText,
+} from '../components';
+
+export default function ArchiveSettings({ isEnabled = false, updateSettings }) {
+  const archiveId = useMemo(() => `archive-${uuidv4()}`, []);
+
+  const onChange = useCallback(
+    () => updateSettings({ archive: !isEnabled }),
+    [updateSettings, isEnabled]
+  );
+
+  const isFeatureEnabled = useFeature('disableArchive');
+
+  if (!isFeatureEnabled) {
```

Comment:
if the feature's not enabled there's no need for the `archiveId` to be defined.

Reply:
archiveId uses the hook `useMemo` and hooks can not be conditional.

Reply:
yup, you're right, read too fast. 👍